app/components/provider_interface/application_choice_header_component.rb

```diff
@@ -56,5 +56,25 @@ def rejection_reason_required
     def offer_present?
       ApplicationStateChange::OFFERED_STATES.include?(application_choice.status.to_sym)
     end
+
+    def respond_to_application?
+      flash.empty? && provider_can_respond && application_choice.awaiting_provider_decision?
+    end
+
+    def deferred_offer_wizard_applicable?
+      flash.empty? && provider_can_respond && deferred_offer_wizard_applicable
+    end
+
+    def rejection_reason_required?
+      flash.empty? && provider_can_respond && rejection_reason_required
+    end
+
+    def show_reject_by_default_text?
```

Comment:
This method name is slightly inconsistent with the above — it describes the behaviour, not the condition. Maybe `provider_cannot_respond?` or sth?